### ISSUE TITLE
no-descending-specificity rule detection

### ISSUE THREAD

**Issue by hiroshi-ishida-246** [stylelint/stylelint#7171]
### What minimal example or steps are needed to reproduce the bug?

```css
.foo > .main > .link {
  color: red;
}

.foo > .link { /* this line "no-descending-specificity" error */
  color: blue;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-descending-specificity": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint 'src/**/*.{css,scss}'
```

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "15.9.0",
    "stylelint-config-recess-order": "4.2.0",
    "stylelint-config-standard": "33.0.0",
    "stylelint-config-standard-scss": "9.0.0",
    "stylelint-order": "6.0.3",
}
```

### What did you expect to happen?

No errors are detected in the example.

### What actually happened?

```
 8:3  ✖  Expected selector ".foobar > .link" to come before selector ".foobar > .main > .link"  no-descending-specificity
```

### Do you have a proposal to fix the bug?

`.foobar > .main > .link` and `.foobar > .link` are different classes.
I think it is necessary to distinguish between the above.

**Comment by ybiquitous** [comment]
@hiroshi-ishida-246 Thanks for using the template with a producible code. I confirmed your report with a [minimal demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEA+DSC2AhgJYB22uANqQNYbAA6ZGkFEATvBmzFANyMBfRo1TpySKiVoMmLdpwBGFAK4x+JISRAAaECiIUYAOQJ44iGAA9TAB0NIwAZ0c7wEEvoDmCEDIwZ6EDZlQ0dAzj9-AJASCABaWEcwGBIoUk84xxsYMCJ9XIAXAE9wjALgmEZ-TQFXSA8iTwAxdkICnwArR3dXWBsXREjAx2LDSQLSwIBGAFYkAGYkAAZAwRABIA) just using CSS (standard).

```css
.foo > .main > .link { /* Specificity: (0, 3, 0) */
  color: red;
}

.foo > .link { /* Specificity: (0, 2, 0) */
  color: blue;
}
```

I'm sorry, but I don't understand what is a problem here. The specificity order in the code above is descending. This is what the [`no-descending-specificity`](https://stylelint.io/user-guide/rules/no-descending-specificity) rule detects.

**Comment by hiroshi-ishida-246** [comment]
@ybiquitous

Thank you for the replay.

> I confirmed your report with a minimal demo just using CSS (standard).

Sorry, I edited the first comment.

> The specificity order in the code above is descending.

I understood that "no-descending-specificity" rule refers to the target name and specificity only just now.

Here's a slightly more practical example.

```css
.header > .list > .item > .link {
  color: red;
}

.footer > .link { /* this line "no-descending-specificity" error */
  color: blue;
}
```

The `.link` in the `.header` and `.link` in the `.footer` are different classes, different target.
The `.footer > .link` cannot override `.header > .list > .item > .link`.

https://stylelint.io/user-guide/rules/no-descending-specificity/
> We recommend turning this rule off if you use a lot of nesting.

It looks like I'll need to turn off the rule for me.

**Comment by Mouvedia** [comment]
I don't think there's anything actionnable here.
If I am mistaken, please free to reopen.
